ProtocolHandler.pm

```diff
@@ -302,8 +302,13 @@ sub getMetadataFor {
 
 	if ($client && $client->playingSong && $client->playingSong->track->url eq $url) {
 		$meta->{bitrate} = $client->playingSong->bitrate if $client->playingSong->bitrate;
-		$meta->{samplerate} = $client->playingSong->pluginData('samplerate') * 1000;
-		$meta->{samplesize} = $client->playingSong->pluginData('samplesize');
+		if ( my $track = $client->playingSong->currentTrack) {
+			$meta->{samplerate} = $track->samplerate if $track->samplerate;
+			$meta->{samplesize} = $track->samplesize if $track->samplesize;
+		} else {
+			$meta->{samplerate} = $client->playingSong->samplerate if $client->playingSong->samplerate;
+			$meta->{samplesize} = $client->playingSong->samplesize if $client->playingSong->samplesize;
+		}
```

Comment:
When would we have a `->currentTrack`, and when not?

Reply:
I can't say exactly but it seems to be created and populated at the time playback begins. Before hitting 'play' for a Qobuz track in the library, the technical info from the status query shows a 'placeholder' bitrate of 750kbps and the sample rate / size of the highest quality Qobuz stream available, as stored in the library. Shortly after starting playback, the average bitrate is calculated using the track's size and duration from the stream info and the sample rate / size values are populated in currentTrack. The values for these attributes in playingSong, however, are not updated. I didn't try to follow the logic down into LMS. The change is admittedly based on empirical observation.

Reply:
I'll trust you'll be tweaking further in case this didn't work 😉. Thanks a lot!